[rocprofiler-sdk] attach: fix test permissions

meserve-amd opened the pull request:
- Test is now skipped if insufficient permissions detected
- Should fix test (for now) in Azure CI pipeline

## Motivation

Not all test systems can be configured to run this test, so we are adding a check for sufficient permissions, and skipping if we believe the test will not be run.

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->

## Technical Details

Sufficient permissions are defined as '/proc/sys/kernel/yama/ptrace_scope' == 0 OR having CAP_SYS_PTRACE OR being run as root. The test will also run if the file does not exist (indicating this distribution does not use ptrace_scope (or it's in a different location, in which case the test should be modified to account for that))

While ctest's fixture system is great, it does not account for skipping FIXTURES_REQUIRED tests if a FIXTURES_SETUP test skipped (which is a legitimate use-case, in the case of optional setup tasks). See discussion [here](https://gitlab.kitware.com/cmake/cmake/-/issues/17980). To skip the `validate` tests, I have the execute test touch an output file called "skipped" which the validate test will check for and skip the test if the file exists.

<!-- Explain the changes along with any relevant GitHub links. -->

## Test Plan

Ran tests locally with /proc/sys/kernel/yama/ptrace_scope set to 0 and 1
Ran tests locally with sudo
Ran tests locally with CAP_SYS_PTRACE set for python3

Use existing CI
<!-- Explain any relevant testing done to verify this PR. -->

## Test Result

Locally, all 6 tests passed when permissions were set, and skipped when they were not.

Azure pipeline CI must pass for this PR.

<!-- Briefly summarize test outcomes. -->

## Submission Checklist

- [x] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

meserve-amd committed:
attach: fix test permissions

- Test is now skipped if insufficient permissions detected
- Should fix test (for now) in Azure CI pipeline

itrowbri commented on projects/rocprofiler-sdk/tests/rocprofv3/attachment/attach-twice/conftest.py:
```diff
@@ -218,19 +220,27 @@ def get_csv_data(file_path):
 @pytest.fixture
 def kernel_input_data(request):
+    if os.path.exists(request.config.getoption("--skip-if")):
+        pytest.skip()
```
Could we add a message about ptrace scope/permissions here? Like `pytest.skip("attach tests unavailable due to insufficient ptrace permissions")` or something along those lines?

meserve-amd replied:
Fixed

itrowbri commented:
Creating a marker file for skipping seems fine to me if everyone else is okay with it as well. I'm going to hold off on approving due to the failing CI for now, but I'll revisit it once it's green/we've addressed most of the test failures in other PRs. Looks like the attach tests are being skipped in the Azure CI summary though, so the everything seems to be working here. We can bring this PR up in the PR review meeting on T and ask about moving it forward anyway since it is a CI fix.

itrowbri reviewed (dismissed):
I'm going to go ahead and approve since this is a CI fix despite the current CI tests failing (failing tests appear to be unrelated to the current PR). Please take a look at my earlier comment about pytrace skip.

meserve-amd committed:
Address review comments

- Add more extensive permission checking for the tests
- Add default parameters to prevent running rm -rf on a root directory
- Add use for unused LOG_LEVEL parameter

itrowbri reviewed (approved):
Approving since it's a CI fix for the Azure pipeline. CI is failing, but doesn't have anything to do with this PR since this PR only touches attach tests.

meserve-amd committed:
Address review comments
